File: algorithms/0500.keyboard-row/solution.rb
# @param {String[]} words
# @return {String[]}
def find_words(words)
  rows = ["qwertyuiop", "asdfghjkl", "zxcvbnm"]
  words.select do |word|
    word_chars = word.downcase.chars

    word_chars.all? {|c| rows[0].include?(c) } ||
      word_chars.all? {|c| rows[1].include?(c) } ||
      word_chars.all? {|c| rows[2].include?(c) }
  end
end

File: algorithms/0500.keyboard-row/solution.rs
impl Solution {
    pub fn find_words(words: Vec<String>) -> Vec<String> {
        let rows = ["qwertyuiop", "asdfghjkl", "zxcvbnm"];

        words
            .into_iter()
            .filter(|word| {
                rows
                    .iter()
                    .fold(false, |same_row, row| {
                        same_row || word.to_lowercase().chars().all(|c| row.contains(c))
                    })
            })
            .collect()
    }
}
